tm2/pkg/db/pebble_db.go

```diff
@@ -0,0 +1,266 @@
+//go:build pebbledb
+
+package db
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+
+	"github.com/cockroachdb/pebble"
+)
+
+const (
+	defaultBytesPerSync = 0 // use the distribution default, 512KB
+	defaultCacheSize    = 0 // use the distribution default, 8MB
+)
+
+func init() {
+	dbCreator := func(name string, dir string) (DB, error) {
+		return NewPebbleDB(name, dir)
+	}
+	registerDBCreator(PebbleDBBackend, dbCreator, false)
+}
+
+var _ DB = (*PebbleDB)(nil)
+
+type PebbleDB struct {
+	db *pebble.DB
+}
+
+func NewPebbleDB(name string, dir string) (*PebbleDB, error) {
+	opts := &pebble.Options{}
+	if defaultBytesPerSync > 0 {
+		opts.BytesPerSync = defaultBytesPerSync
+	}
+	if defaultCacheSize > 0 {
+		cache := pebble.NewCache(defaultCacheSize)
+		defer cache.Unref()
+		opts.Cache = cache
+	}
+	return NewPebbleDBWithOpts(name, dir, opts)
+}
+
+func NewPebbleDBWithOpts(name string, dir string, opts *pebble.Options) (*PebbleDB, error) {
+	db, err := pebble.Open(filepath.Join(dir, name+".db"), opts)
+	if err != nil {
+		return nil, err
+	}
+	return &PebbleDB{
+		db: db,
+	}, nil
+}
+
+// Implements DB.
+func (db *PebbleDB) Get(key []byte) []byte {
+	value, closer, err := db.db.Get(nonNilBytes(key))
+	if err != nil {
+		if errors.Is(err, pebble.ErrNotFound) {
+			return nil
+		}
+		panic(err)
+	}
```

Comment:
Avoid unnecessary nesting:

```suggestion
	if errors.Is(err, pebble.ErrNotFound) {
		return nil
	}
	if err != nil {
		panic(err)
	}
```

Reply:
I think for most of cases where err has nil value, the suggested logic will cost more as it goes through both L57 and L60.

Reply:
> I honestly don't see why the suggestion is a better way.

It is easier to read. If the code path is so hot that an extra `if` check will affect performance meaningfully, should we add some benchmarks to be careful about that?

Reply:
Certainly, prioritizing readability and simplicity is beneficial, even in smaller cases. By consistently emphasizing clarity, we foster good habits and ensure our work remains accessible to others. It's a commendable practice to always strike a balance between efficiency and readability.

See https://github.com/gnolang/gno/pull/1111